packages/router/src/adapters/contract/contract.ts

```diff
@@ -483,122 +571,24 @@ export const fulfillRouterContract = async (
-      );
-      if (!data.taskId) {
-        throw new Error("No taskId returned");
-      }
-      logger.info("Router contract fulfill: sent using Gelato relayer", requestContext, methodContext, { data });
-
-      // listen for event on contract
-      const { event } = await fulfillEvt
-        .pipe(({ args }) => args.txData.transactionId === txData.transactionId)
-        .waitFor(300_000);
-
-      // increment router fees
-      incrementRelayerFeesPaid(
-        chainId,
-        routerRelayerFee,
-        routerRelayerFeeAsset,
-        TransactionReasons.FulfillSender,
-        requestContext,
-      );
-      return await txService.getTransactionReceipt(chainId, event.transactionHash);
-    } catch (err: any) {
-      logger.warn("Router contract fulfill: Gelato send failed", requestContext, methodContext, {
-        err: jsonifyError(err),
-      });
-    }
-  }
-
-  // 2. If gelato is not supported, or gelato send failed, try using the router network.
-  if (useRelayer) {
-    logger.info("Router contract fulfill: sending using router network", requestContext, methodContext, {
-      fulfillParams,
-      routerRelayerFeeAsset,
-      routerRelayerFee,
-    });
-
-    try {
-      const payload = {
-        chainId,
-        to: routerContractAddress,
-        type: MetaTxTypes.RouterContractFulfill,
-        data: {
-          params: fulfillParams,
-          signature,
-          relayerFee: routerRelayerFee,
-          relayerFeeAsset: routerRelayerFeeAsset,
-        } as MetaTxPayloads[typeof MetaTxTypes.RouterContractFulfill],
-      };
-      await messaging.publishMetaTxRequest(payload);
-
-      // listen for event on contract
-      const { event } = await fulfillEvt
-        .pipe(({ args }) => args.txData.transactionId === txData.transactionId)
-        .waitFor(300_000);
-
-      // increment router fees
-      incrementRelayerFeesPaid(
-        chainId,
-        routerRelayerFee,
-        routerRelayerFeeAsset,
-        TransactionReasons.FulfillSender,
-        requestContext,
-      );
-
-      return await txService.getTransactionReceipt(chainId, event.transactionHash);
-    } catch (err: any) {
-      // NOTE: It is possible that the actual error was in the subscriber, and the above event's timeout
-      // (see waitFor) is the error we actually caught in this block.
-      logger.warn("Router contract fulfill: router network failed", requestContext, methodContext, {
-        err: jsonifyError(err),
-      });
-    }
-  }
-
-  // 3. If all of the above failed or was otherwise not supported, use txservice to send the transaction.
-  logger.info("Router contract fulfill: sending using txservice", requestContext, methodContext, { fulfillParams });
-  const receipt = await txService.sendTx(
-    onchainTx,
+      relayerFee: useRelayer
+        ? {
+            asset: routerRelayerFeeAsset,
+            amount: routerRelayerFee,
+          }
+        : undefined,
+      reason: TransactionReasons.FulfillSender,
```

Comment:
Could still be a `metaTx` relay, as seen [here](https://github.com/connext/nxtp/blob/cdaccfdc3018803e2df6bba26e6b77f969de6582/packages/router/src/lib/operations/metaTx.ts#L205)